.github/workflows/conformance-delegated-ipam.yaml

```diff
@@ -0,0 +1,156 @@
+name: Conformance Delegated IPAM
+
+# TODO: update this to run on a schedule.
+on: workflow_dispatch
```

Comment:
Is there an example pipeline I can copy for the trigger to use for a post-merge test? I saw some pipelines that used "scheudle"

Reply:
Currently, we don't run all workflows on a schedule, only a selection of them. Most of them are workflows that have been passing reliably for some time. So I'd suggest to first add this workflow without a schedule and once the workflow proves stable on PR runs add an optional schedule if needed.

Reply:
Updated this to match conformance-multi-pool.yaml. Looks like that runs on pushes to the main branch, which I think is what we want for post-merge tests like @joestringer suggested https://github.com/cilium/cilium/issues/34788#issuecomment-2338722420

Reply:
Passing on this run I triggered from my fork: https://github.com/wedaly/cilium/actions/runs/10836785242/job/30071291776

I can see that:
1. cilium install is using delegated IPAM parameters.
2. kind cluster is created with three nodes, none of which are assigned a pod CIDR by controller-manager.
3. nodes become ready 
4. coredns pods are assigned pod IP from 10.244.1.0/24 (the the subnet configured for host-local IPAM plugin on the control plane node).
4. cilium connectivity tests pass, which means pods were assigned IPs on the worker nodes as well.

Reply:
Hmm yeah I must admit I didn't think too much about how to stabilize the workflow in the first place. Broadly I don't think this needs to be part of testing for every PR, as it's likely to be rare that delegated IPAM breaks. Given the background of #34688 , I'm trying to encourage that we consider both the benefit and the impact that new tests will have, and avoid assuming every PR needs to run every test. What I'd like to see with this is if we think that something touches delegated ipam, we can do a `/test-delegated-ipam`, and otherwise we just run it either on pushes to `main` or on a schedule. Schedule might add some complication though, as I recall we have hit issues in the past with scheduling too many jobs at once and causing failures that way.

That said, maybe running as a non-required test on every PR and monitoring the reliability of the tests via https://github.com/cilium/cilium/actions could reduce the time-to-stability for a few weeks, then we could take it out of the per-PR testing.